src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -206,12 +534,48 @@ public SDKRestClient restClient() {
         return this.sdkRestClient;
     }
 
-    // @Override
-    public Map<String, Class<? extends TransportAction<? extends ActionRequest, ? extends ActionResponse>>> getActionsMap() {
-        Map<String, Class<? extends TransportAction<? extends ActionRequest, ? extends ActionResponse>>> map = new HashMap<>();
-        map.put(ADJobParameterAction.NAME, ADJobParameterTransportAction.class);
-        map.put(ADJobRunnerAction.NAME, ADJobRunnerTransportAction.class);
-        return map;
+    /**
+     * createComponents doesn't work for Clock as ES process cannot start
+     * complaining it cannot find Clock instances for transport actions constructors.
+     * @return a UTC clock
+     */
+    protected Clock getClock() {
+        return Clock.systemUTC();
+    }
+
+    @Override
+    public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
+        return ImmutableList
+            .of(
+                new ScalingExecutorBuilder(
+                    AD_THREAD_POOL_NAME,
+                    1,
+                    // HCAD can be heavy after supporting 1 million entities.
+                    // Limit to use at most half of the processors.
+                    Math.max(1, OpenSearchExecutors.allocatedProcessors(settings) / 2),
+                    TimeValue.timeValueMinutes(10),
+                    AD_THREAD_POOL_PREFIX + AD_THREAD_POOL_NAME
+                )// ,
```

Comment:
(Style) Would really prefer block comment format-preserving comments here.  (Start with `/*-` and keep the indentation in the middle before the `*/`.

Reply:
Ill go ahead and fix this